spec/features/course/achievement_management_spec.rb

```diff
@@ -11,6 +11,40 @@
       login_as(user, scope: :user)
     end
 
+    def add_achievement_to_course(achievement, course)
+      course.tap do |c|
+        c.achievements << achievement
+        c.save
+      end
+    end
+
+    def create_achievement_in_course(course)
+      create(:course_achievement, course: course).tap do |a|
+        add_achievement_to_course(a, course)
+      end
+    end
+
+    def create_achievement_condition_stubs
+      new_course = create(:course)
```

Comment:
I think this is unnecessarily complicated because you aren't using the create factories correctly. You should be able to directly create achievements and courses without needing to write such methods.